Automattic/wp-calypso
# Stats: Update Devices module with new empty state

Related to https://github.com/Automattic/red-team/issues/70

## Proposed Changes

* update the Devices module with a new empty state and place it behind a feature flag
* refactor TS definitions
* add the new skeleton loader to the module

## Why are these changes being made?


* a part of the empty states project

## Testing Instructions



* go to the live branch
* apply `stats/empty-module-traffic` feature flag
* verify that a page with an empty Devices component works with and without the feature flag
* repeat for blogs with data
* verify that the advanced feature overlay works properly
* verify that the new empty state only shows up after applying the feature flag

![SCR-20240708-lhvl](https://github.com/Automattic/wp-calypso/assets/112354940/fe3d31ad-00a4-4be7-8ea6-5cf8d5b27b8f)

## Pre-merge Checklist
